docs/quickstart.md

```diff
@@ -213,6 +219,80 @@ info: Echo Handler: teardown method called.
 info: Echo Behaviour: teardown method called.
 ```
 
+## Write a test for the AEA
+
+We can write an end-to-end test for the AEA utilising helper classes provided by the framework.
+
+<details><summary>Step by step install</summary>
+
+The following test class replicates the preceding demo and tests it's correct behaviour. The `AEATestCase` is a tool for AEA developers to write useful end-to-end tests of their AEAs.
+
+``` python
+import os
+import signal
+import time
+
+from aea.connections.stub.connection import (
+    DEFAULT_INPUT_FILE_NAME,
+    DEFAULT_OUTPUT_FILE_NAME,
+)
+from aea.mail.base import Envelope
+from aea.protocols.default.message import DefaultMessage
+from aea.protocols.default.serialization import DefaultSerializer
+from aea.test_tools.decorators import skip_test_ci
+from aea.test_tools.generic import (
+    read_envelope_from_file,
+    write_envelope_to_file,
+)
+from aea.test_tools.test_cases import AEATestCase
+
+
+class TestEchoSkill(AEATestCase):
```

Comment:
@marcofavorito and @panasevychol we would like to make this AEATestCase useful for developers. But actually, if you look at the below code taken from tests, we still lack a few things on the test case. So in the tests it makes sense to create the whole project from scratch. But here it would be preferable to lose lines 256-263 and instead point to the project directory which already exists and run the test. What do you think?

Reply:
yes in my opinion. Maybe add a comment like:
```python
# we assume the AEA project has already been set up 
agent_dir_path = agent_name
os.chdir(agent_dir_path)
```

Reply:
The problem is, the test case does not support it atm, so we need to extend it.

Reply:
Have a look at the new implementation @marcofavorito